### PR TITLE
Fix MTsat-generated T1 value bug when CLI is used with JSON sidecar

### LINKED ISSUE #3810: Wrong T1 map
We've recently discovered T1 maps which value are off by ~3 orders of magnitude:
- https://github.com/spine-generic/spine-generic/issues/257
- https://github.com/courtois-neuromod/anat-processing-book/pull/39

There is something fishy in the code, notably:

https://github.com/spinalcordtoolbox/spinalcordtoolbox/blob/9d367a487c5d05e22c13ee70b24197ad399a12bf/spinalcordtoolbox/qmri/mt.py#L90-L93

And here:
https://github.com/spinalcordtoolbox/spinalcordtoolbox/blob/9d367a487c5d05e22c13ee70b24197ad399a12bf/spinalcordtoolbox/qmri/mt.py#L104-L118

Dataset for testing:
~~~
sct_download_data -d sct_testing_data
~~~

While fixing this bug, we should also mention the expected input unit:
https://github.com/spinalcordtoolbox/spinalcordtoolbox/blob/9d367a487c5d05e22c13ee70b24197ad399a12bf/spinalcordtoolbox/qmri/mt.py#L70-L72

### PR DESCRIPTION
**Pull request by mathieuboudreau**
#### GitHub

- [x] I've given this PR a concise, self-descriptive, and meaningful title
- [x] I've linked relevant issues in the PR body
- [x] I've applied [the relevant labels](https://intranet.neuro.polymtl.ca/geek-tips/contributing#pr-labels-a-href-pr-labels-id-pr-labels-a) to this PR
- [ ] I've applied a [release milestone](https://github.com/spinalcordtoolbox/spinalcordtoolbox/milestones) (major, minor, patch) in line with [Semantic Versioning guidelines](https://github.com/spinalcordtoolbox/spinalcordtoolbox/wiki/Misc%3A-Creating-a-new-release#convention-for-naming-releases) 
- [x] I've assigned a reviewer

#### PR contents

- [x] I've consulted [SCT's internal developer documentation](https://github.com/spinalcordtoolbox/spinalcordtoolbox/wiki) to ensure my contribution is in line with any relevant design decisions
- [x] I've added [relevant tests](https://github.com/spinalcordtoolbox/spinalcordtoolbox/wiki/Programming%3A-Tests) for my contribution
- [x] I've updated the [relevant documentation](https://github.com/spinalcordtoolbox/spinalcordtoolbox/wiki/Programming%3A-Documentation) for my changes, including argparse descriptions, docstrings, and ReadTheDocs tutorial pages

## Description

Resolves #3810. T1 values generated when CLI was used with JSON sidecars resulted in T1 values 1000x smaller than true value, due to difference in input repetition times to the API (expected ms, was getting seconds through JSON). Adds tests that would have caught the issue.

## Linked issues
#3810

### CODE REVIEW

**Comment by mathieuboudreau** [comment]
Now that I've written tests that fail for the current implementation, here comes the fix.

**Comment by mathieuboudreau** [comment]
Now that we have expected behaviour tests and a bug fix that resolved them, I'll refactor the code to change the TR inputs from ms to seconds, per the discussion here: https://github.com/spinalcordtoolbox/spinalcordtoolbox/issues/3810#issuecomment-1158953392

Repeating @jcohenadad 's comment, 

>If we introduce this change, we would need to alert users on the next release. Tagging @joshuacwnewton @mguaypaq

File: testing/cli/test_cli_sct_compute_mtsat.py [hunk 1]
```diff
@@ -21,14 +26,32 @@
      '-t1', sct_test_path('mt', 't1w.nii.gz'),
      '-omtsat', out_mstat,
      '-ot1map', out_t1map,
-     '-trmt', '51', '-trpd', '52', '-trt1', '10', '-famt', '4', '-fapd', '5', '-fat1', '14'],
+     '-trmt', '0.030', '-trpd', '0.030', '-trt1', '0.015', '-famt', '9', '-fapd', '9', '-fat1', '15'],
```
Comment (mguaypaq) [inline testing/cli/test_cli_sct_compute_mtsat.py:29]: Thanks for making these values consistent with the JSON sidecar.

**Review by mguaypaq — APPROVED** [review]
Thanks for the investigation, the new test, the fix, and updating the documentation!

I believe you already fixed all the places which call `compute_mtsat`. There was also another call to `sct_compute_mtsat` in `batch_processing.py`, so I added a commit to fix it. I also merged the two value tests into one, since the same computation generates both tested values.

The BIDS specification does indeed say that [RepetitionTime is measured in seconds](https://bids-specification.readthedocs.io/en/stable/99-appendices/14-glossary.html#repetitiontime-metadata).

Since you had a couple of commits to fix linting errors, I'll note the following tip: you can set up your development environment to [automatically run the same linting step](https://github.com/spinalcordtoolbox/spinalcordtoolbox/wiki/Contributing%3A-Installing-SCT-for-development#step-5-linting-automatically) on your computer as happens on Github, that way you get warned as soon as you run `git commit`.

I think @jcohenadad already expressed his opinions on the linked issue #3810, so feel free to merge this PR.

File: spinalcordtoolbox/scripts/sct_compute_mtsat.py [hunk 2]
```diff
@@ -61,19 +61,19 @@ def get_parser():
         help="Show this help message and exit")
     optional.add_argument(
         "-trmt",
-        help="TR [in ms] for mt image. By default, will be fetch from the json sidecar (if it exists).",
+        help="TR [in s] for mt image. By default, will be fetch from the json sidecar (if it exists).",
```
Comment (jcohenadad) [inline spinalcordtoolbox/scripts/sct_compute_mtsat.py:64]:
```suggestion
        help="TR [in s] for the MT image (MT on). By default, will be fetch from the json sidecar (if it exists).",
```

File: spinalcordtoolbox/scripts/sct_compute_mtsat.py [hunk 3]
```diff
@@ -61,19 +61,19 @@ def get_parser():
...
         type=float,
         metavar=Metavar.float,
     )
     optional.add_argument(
         "-trpd",
-        help="TR [in ms] for pd image. By default, will be fetch from the json sidecar (if it exists).",
+        help="TR [in s] for pd image. By default, will be fetch from the json sidecar (if it exists).",
```
Comment (jcohenadad) [inline spinalcordtoolbox/scripts/sct_compute_mtsat.py:70]:
```suggestion
        help="TR [in s] for proton density weighted image (MT off). By default, will be fetch from the json sidecar (if it exists).",
```

File: spinalcordtoolbox/scripts/sct_compute_mtsat.py [hunk 4]
```diff
@@ -61,19 +61,19 @@ def get_parser():
...
         type=float,
         metavar=Metavar.float,
     )
     optional.add_argument(
         "-trt1",
-        help="TR [in ms] for t1 image. By default, will be fetch from the json sidecar (if it exists).",
+        help="TR [in s] for t1 image. By default, will be fetch from the json sidecar (if it exists).",
```
Comment (jcohenadad) [inline spinalcordtoolbox/scripts/sct_compute_mtsat.py:76]:
```suggestion
        help="TR [in s] for T1-weighted image. By default, will be fetch from the json sidecar (if it exists).",
```

File: testing/cli/test_cli_sct_compute_mtsat.py (hunk 1 above)
Comment (mathieuboudreau) [reply testing/cli/test_cli_sct_compute_mtsat.py:29]: No problem!

**Pull request merged**